pdns/pdns_recursor.cc

```diff
@@ -4014,6 +4014,13 @@ static int serviceMain(int argc, char*argv[])
   g_numThreads = g_numDistributorThreads + g_numWorkerThreads;
   g_maxMThreads = ::arg().asNum("max-mthreads");
 
+  
+  if (TCPConnection::s_maxInFlight < 1 || TCPConnection::s_maxInFlight > USHRT_MAX || TCPConnection::s_maxInFlight >= g_maxMThreads) {
+    g_log<<Logger::Warning<<"Asked to run with illegal max-concurrent-requests-per-tcp-connection, setting to default"<<endl;
```

Comment:
You might want to log the default as well :)

Reply:
Oh, how I wish the default could be accessed by code ....

Reply:
Actually, the logic is wrong. I need to validate before assigning to a ushort.